tests/FixerFactoryTest.php

```diff
@@ -380,15 +379,38 @@ public function testFixerDefinitions(FixerInterface $fixer)
             $this->markTestIncomplete('ShortFixerDefinition does not contains all needed information.');
         }
 
-        $this->assertNotEmpty($definition->getCodeSamples(), 'Code samples are required.');
+        $samples = $definition->getCodeSamples();
+        $this->assertInternalType('array', $samples, 'Code samples must be an array.');
+        $this->assertNotEmpty($samples, 'Code samples are required.');
+
+        foreach ($samples as $sample) {
+            $this->assertInstanceOf('PhpCsFixer\FixerDefinition\CodeSampleInterface', $sample);
+            $code = $sample->getCode();
+            $this->assertStringIsNotEmpty($code);
+            $config = $sample->getConfiguration();
+            if (null !== $config) {
+                $this->assertInternalType('array', $config, 'Sample config must be an array or null.');
+            }
+
+            // TODO
+            // test code is valid PHP
+            // test code is candidate for fixer
```

Comment:
execute the `$fixer->fix` and see if input is different than output.
in the meantime, Tokenizer will raise an exception for invalid PHP on PHP7